portfolio/src/test/java/com/google/sps/NumCommentsTest.java

```diff
@@ -0,0 +1,200 @@
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+  
+  /*
+   * Helper that simply removes all whitespace and invisible chars (tabs, \n...)
+   */
+  public String removeWhitespace(String input) {
+    return input.replaceAll("\\s+","");
+  }
+
+  /*
+   * Turn a response StringWriter into an Int
+   */
+  public int StringWriterToInt(StringWriter stringWriter) {
+    String string = stringWriter.toString();
+    String withoutWhitespace = removeWhitespace(string);
+    try {
+      int inputValue = Integer.valueOf(removeWhitespace(string));
+      return inputValue;
+    } catch (NumberFormatException nfe) {
+      throw new NumberFormatException("Couldn't parse " + withoutWhitespace);
+    }
+  } 
+
+  /*
+   * Just make a basic comment entity for testing purposes.
+   */
+  public Entity basicComment() {
+    Entity commentEntity = new Entity("Comment");
+    Long oneLong = Long.valueOf(1);
+
+    commentEntity.setProperty("comment", "");
+    commentEntity.setProperty("timestamp", oneLong);
+    commentEntity.setProperty("upvoteCount", oneLong);
+    commentEntity.setProperty("id", oneLong);
+    return commentEntity;
+  }
+
+  /*
+   * 
+   */
+  public StringWriter getStringWriter(HttpServletResponse response) throws Exception {
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+    
+    return stringWriter;
+  }
+
+  @Test
+  public void testEmpty() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    
+    HttpServletRequest request = mock(HttpServletRequest.class);       
+    HttpServletResponse response = mock(HttpServletResponse.class); 
+
+    StringWriter stringWriter = getStringWriter(response);
+
+    new NumCommentsServlet().doGet(request, response);
+    
+    int numComments = StringWriterToInt(stringWriter);
+    Assert.assertEquals(numComments, 0);
+  }
+
+  @Test 
+  public void testOneComment() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    
+    Entity commentEntity = basicComment();
+    ds.put(commentEntity);
+
+    HttpServletRequest request = mock(HttpServletRequest.class);       
+    HttpServletResponse response = mock(HttpServletResponse.class); 
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    new NumCommentsServlet().doGet(request, response);
+        
+    int numComments = StringWriterToInt(stringWriter);
+    Assert.assertEquals(numComments, 1);
+  }
+  
+  @Test
+  public void testAddedCommentsCounted() throws Exception {    
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+
+    Entity commentEntity1 = basicComment();
+    ds.put(commentEntity1);
+
+    HttpServletRequest request = mock(HttpServletRequest.class);       
```

Comment:
A) name your mock variables mockWhatever
B) Usually you don't declare mocks in the test itself.  You declare the mocks as class variables for the test class.  They should automatically be reset after every individual test method runs.

Reply:
Why are you mocking these anyway?  You never actually use the mocks for anything.